### DISCUSSION TITLE
`set(obj, key, undefined)` should set the value to undefined

### DISCUSSION THREAD

**Discussion by AndreyYolkin** [radashi-org/radashi#296] [Ideas]
We are investigating the ways we could safely migrate from lodash-es and use modern library with better state, support and code quality.

The only blocker we have now is the difference between set() behavior in lodash and radashi

Let's say we have object with server errors after validation:

```ts
const errors = {
  email: 'Should include @',
  entries: [{ date: 'Should be date in YYYY-MM-DD' }]
}
```

then i want to strip error after user provided correct date via `set(errors, 'entries[0].date', undefined)`

Lodash version sets value as undefined, while radashi makes early return, leaving the value untouched

<img width="465" alt="image" src="https://github.com/user-attachments/assets/bb97284f-30b9-4d8b-abc5-3598fe232a10">

I suggest to make parity between behavior in lodash and radashi in that particular case. If `undefined` [check](https://github.com/radashi-org/radashi/blob/28c6efcf8b19227573c4f9a325ef47303c314827/src/object/set.ts#L23) had been designed to just check if 3rd argument was passed, we can check arguments length instead.

I also understand this is going to be a breaking change, so we this may happen in 13.0

**Comment by aleclarson** [comment]
I approve. We're already planning on removing other nullish checks from the codebase: https://github.com/radashi-org/radashi/pull/70

**Reply by AndreyYolkin** [reply]
That's great! Looks like we can remove this check https://github.com/radashi-org/radashi/blob/a6d708e53ff55aafd289d6508fadb12141afed1a/src/path/set.ts#L17 completely and make function even shorter. Will wait for RC to try!
